nym-vpn-core/crates/nym-vpn-store/src/helpers.rs

```diff
@@ -0,0 +1,107 @@
+use std::path::{Path, PathBuf};
+
+use rand::SeedableRng;
+
+use crate::{key_store::KeyStore, DeviceKeys, DeviceKeysPaths, OnDiskKeys, OnDiskKeysError};
+
+#[derive(Debug, thiserror::Error)]
+pub enum KeyStoreError {
+    #[error("failed to load device keys")]
+    Load {
+        path: PathBuf,
+        error: OnDiskKeysError,
+    },
+
+    #[error("failed to create device keys")]
+    Create {
+        path: PathBuf,
+        error: OnDiskKeysError,
+    },
+
+    #[error("failed to store device keys")]
+    Store {
+        path: PathBuf,
+        error: OnDiskKeysError,
+    },
+
+    #[error("invalid key pair, one is missing: {path}")]
+    InvalidKeyPair { path: PathBuf },
+}
+
+pub fn keypair_exists<P: AsRef<Path>>(path: P) -> Result<bool, KeyStoreError> {
+    let device_key_paths = DeviceKeysPaths::new(path);
+    let private_key_exists = device_key_paths.private_device_key().exists();
+    let public_key_exists = device_key_paths.public_device_key().exists();
+    match (private_key_exists, public_key_exists) {
+        (true, true) => Ok(true),
+        (false, false) => Ok(false),
+        (true, false) => Err(KeyStoreError::InvalidKeyPair {
```

Comment:
that imo shouldn't return an error. sure, we could log a warning here, but we could just load the private key and trivially derive the public key out of it. think of ssh keys. you don't really store public keys at all. you just persist the private one and generate public on demand

Reply:
Good point, I'll change it